src/index.ts

```diff
@@ -980,6 +991,59 @@ class Envoy {
     }
   }
 
+
+  // --------------------------------------------------------------------------------------------
+
+  // Functions handling translation of dna_hash to hha_hash
+
+  async recordHha(hha_hash) {
+    // dna2hha is add-only
+    if (!this.hhaExists(hha_hash)) {
+      log.info("Retrieve the hosted app cell_data using the anonymous installed_app_id: '%s'", hha_hash);
+
+      const appInfo = await this.callConductor("app", { installed_app_id: hha_hash });
+
+      if (!appInfo) {
+        throw new Error(`No app found with installed_app_id: ${hha_hash}`);
+      }
+
+      // TODO: I am operating under the assumption that each dna_hash can be only in one app (identified by hha_hash)
+      // Does this need to change?
+      appInfo.cell_data.forEach(cell => {
+        this.dna2hha[cell[0][0]] = hha_hash;
+      });
+    }
+  }
+
+  hhaExists(hha_hash) {
+    return (Object.values(this.dna2hha).includes(hha_hash));
+  }
+
+  async signalHandler(signal) {
+    let cell_id = signal.data.cellId;
+    log.debug("Received signal for cellId (%s)", cell_id);
+
+    // translate CellId->eventId
+    let event_id = this.cellId2eventId(cell_id);
+
+    log.debug(`Emitting 'signal' to event ${event_id}:`);
+    log.debug(signal);
+    this.ws_server.emit(event_id, signal)
+  }
+
+  cellId2eventId(cell_id) {
+    // CellId is of a format `agent_id:dna_hash`
+    let cell_arr = cell_id.split(":");
```

Comment:
The signal structure coming from holochain-conductor-api looks like this
```
export type AppSignal = {
  type: string,
  data: {
    cellId: CellId,
    payload: any
  }
}

export type CellId = [
    // DNA hash
    HoloHash,
    AgentPubKey
]
```
So I don't think cell ID's are of that format with a colon. Maybe you're thinking of cell nicknames?

Reply:
I bet you're thinking of `installed-app-id`s

Reply:
updated